src/hooks/useInitialValue.ts

```diff
@@ -0,0 +1,9 @@
+import {useState} from 'react';
+
+// In some places we set initial value on first render, but we don't want to re-run the function
+// This hook will memoize the initial value and return that without setter, so it's never changed
+// https://github.com/Expensify/App/pull/29643#issuecomment-1765894078
+export default function useInitialValue<T>(initialStateFunc: () => T) {
+    const [initialValue] = useState(initialStateFunc);
```

Comment:
@allroundexperts I'm passing function to `useState` and useState will run it only once for computing initial value. It's used that way to avoid unnecessary recalculation of initial state function, as described in [react docs](https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state) (calling `useState(initialStateFunc())` will run `initialStateFunc` on each render).